# General Vision

**This master thesis will investigate LC-PUFA biosynthesis pathways in Canuelloida during different temperature regimes simulating the change in the natural environment occurring due to seawater warming. The objective of this master thesis will include contributing to the currently available transcriptome data to help infer phylogenetic relationships and LC-PUFA biosynthesis-related genes in Canuelloida.**

## Thesis vision
By the end of this thesis, I would like to have an understanding of the LC-PUFA biosynthesis in Canuelloida.

---
## Thesis objectives

### Master thesis objectives

The main objective is to define LC-PUFA biosynthesis in Canuella perplexa and the related genes under different temperature ranges while contributing to the currently available transcriptome data to help infer phylogenetic relationships of the Canuelloida.

### Specific objectives

1. To quantify fatty acids found in Canuella and their food under different temperature regimes (broad temperature range).

2. To identify genes coding for the biosynthesis pathway and quantify differential gene expression under different temperatures.

3. To analyze correlations between fatty acid concentration and gene expression levels.

---
## Milestones

### 1. Fatty acids quantification

1. Fatty acid identification via Masshunter

2. Fatty acid quantification via Massquant and excel

3. Statistical analysis to infer patterns of FA composition in the data



### 2. RNA-Seq Analysis

![Read mapping and transcript identification strategies](/Users/nabwireasatsa/Pictures/Photos Library.photoslibrary/originals/B/B512547F-5570-47F2-A4A6-00D935A5FC9F.jpeg)

In this project we will make use of strategy "C" this is because we do not have a reference transcriptome.

1. QC of the raw data (paired end sequence data from the illumina plattform)
Tools
- FastQC
2. Trimming low quality reads and removing adapters if any
Tools:

- Trimmomatic- can be used to discard low-quality reads, trimadaptor sequences, and eliminate poor-quality bases

3. Read alignment

I will be performing both de novo assembly and also make use of a reference trancsriptome.

The GC content of mapped reads may reveal PCR biases. Tools for quality control in mapping include Picard, RSeQC and Qualimap.

RNA-seq reads can be assembled de novo into a transcriptome using SOAPdenovo-Trans, Oases, Trans-ABySS or Trinity

4. Transcript quantification

This will aim to estimate gene and transcript expression

5. Differential gene expression analysis

6. Visualization

### 3.Integration with other data types

The integration of RNA-seq data with other types of genome wide data allows us to connect ther egulation of gene expression with specific aspects of molecular physiology and functional genomics. We will integrate FA analysis results with RNA-Seq data.

### 4. Thesis writing

### 5. Preparation of manuscript for publishing
